### PR TITLE
Adding the specs for GA version of Events API

### PR DESCRIPTION
**Pull request by srinathgrs**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/ResourceHealth.json [hunk 1]
```diff
@@ -0,0 +1,1239 @@
...
+                "Unplanned",
+                "Planned",
+                "UserInitiated"
+              ],
+              "x-ms-enum": {
+                "name": "ReasonTypeValues",
+                "modelAsString": true
+              }
+            },
+            "occuredTime": {
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/ResourceHealth.json]:
```suggestion
            "occurredTime": {
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/ResourceHealth.json [hunk 2]
```diff
@@ -0,0 +1,1239 @@
...
+            "healthEventId": {
+              "type": "string",
+              "description": "It is a unique Id that identifies the event"
+            },
+            "resolutionETA": {
+              "type": "string",
+              "description": "When the resource's availabilityState is Unavailable and the reasonType is not User Initiated, it provides the date and time for when the issue is expected to be resolved.",
+              "format": "date-time"
+            },
+            "occuredTime": {
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/ResourceHealth.json]:
```suggestion
            "occurredTime": {
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatus_GetByResource.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        "name": "current",
+        "type": "Microsoft.ResourceHealth/AvailabilityStatuses",
+        "location": "eastus",
+        "properties": {
+          "availabilityState": "Unavailable",
+          "summary": "We're sorry, we couldn't automatically recover your virtual machine",
+          "reasonType": "Unplanned",
+          "reasonChronicity": "Persistent",
+          "detailedStatus": "Disk problems are preventing us from automatically recovering your virtual machine",
+          "occuredTime": "2016-03-29T09:12:00Z",
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatus_GetByResource.json]:
```suggestion
          "occurredTime": "2016-03-29T09:12:00Z",
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_List.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            "name": "current",
+            "type": "Microsoft.ResourceHealth/AvailabilityStatuses",
+            "location": "eastus",
+            "properties": {
+              "availabilityState": "Unavailable",
+              "summary": "We're sorry, we couldn't automatically recover your virtual machine",
+              "reasonType": "Unplanned",
+              "reasonChronicity": "Persistent",
+              "detailedStatus": "Disk problems are preventing us from automatically recovering your virtual machine",
+              "occuredTime": "2016-03-29T09:12:00Z",
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_List.json]:
```suggestion
              "occurredTime": "2016-03-29T09:12:00Z",
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_List.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            "id": "/subscriptions/4abcdefgh-ijkl-mnop-qrstuvwxyz/resourceGroups/rhctestenv/providers/Microsoft.ClassicCompute/virtualMachines/rhctes tenvV1PI/providers/Microsoft.ResourceHealth/availabilityStatuses/2016-03-28+16%3a23%3a00Z",
+            "name": "2016-03-28+16%3a23%3a00Z",
+            "type": "Microsoft.ResourceHealth/AvailabilityStatuses",
+            "location": "eastus",
+            "properties": {
+              "availabilityState": "Available",
+              "summary": "This virtual machine is running normally",
+              "reasonChronicity": "Persistent",
+              "detailedStatus": "There aren’t any known Azure platform problems affecting this virtual machine",
+              "occuredTime": "2016-03-28T16:23:00Z"
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_List.json]:
```suggestion
              "occurredTime": "2016-03-28T16:23:00Z"
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_ListByResourceGroup.json [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            "name": "current",
+            "type": "Microsoft.ResourceHealth/AvailabilityStatuses",
+            "location": "eastus",
+            "properties": {
+              "availabilityState": "Available",
+              "summary": "Vm is available",
+              "reasonType": "Unplanned",
+              "reasonChronicity": "Persistent",
+              "detailedStatus": "We have not seen any issues with your virtual machine",
+              "occuredTime": "2016-03-29T09:12:00Z",
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_ListByResourceGroup.json]:
```suggestion
              "occurredTime": "2016-03-29T09:12:00Z",
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_ListByResourceGroup.json [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            "name": "current",
+            "type": "Microsoft.ResourceHealth/AvailabilityStatuses",
+            "location": "eastus",
+            "properties": {
+              "availabilityState": "Unavailable",
+              "summary": "We are sorry, we couldn't automatically recovery our virtualmachine",
+              "reasonType": "Unplanned",
+              "reasonChronicity": "Persistent",
+              "detailedStatus": "Diskproblemsarepreventingusfromautomaticallyrecoveringyourvirtualmachine",
+              "occuredTime": "2016-03-29T09:12:00Z",
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_ListByResourceGroup.json]:
```suggestion
              "occurredTime": "2016-03-29T09:12:00Z",
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_ListBySubscriptionId.json [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            "name": "current",
+            "type": "Microsoft.ResourceHealth/AvailabilityStatuses",
+            "location": "eastus",
+            "properties": {
+              "availabilityState": "Available",
+              "summary": "Vm is available",
+              "reasonType": "Unplanned",
+              "reasonChronicity": "Persistent",
+              "detailedStatus": "We have not seen any issues with your virtual machine",
+              "occuredTime": "2016-03-29T09:12:00Z",
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_ListBySubscriptionId.json]:
```suggestion
              "occurredTime": "2016-03-29T09:12:00Z",
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_ListBySubscriptionId.json [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            "name": "current",
+            "type": "Microsoft.ResourceHealth/AvailabilityStatuses",
+            "location": "eastus",
+            "properties": {
+              "availabilityState": "Unavailable",
+              "summary": "We are sorry, we couldn't automatically recovery our virtualmachine",
+              "reasonType": "Unplanned",
+              "reasonChronicity": "Persistent",
+              "detailedStatus": "Diskproblemsarepreventingusfromautomaticallyrecoveringyourvirtualmachine",
+              "occuredTime": "2016-03-29T09:12:00Z",
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/AvailabilityStatuses_ListBySubscriptionId.json]:
```suggestion
              "occurredTime": "2016-03-29T09:12:00Z",
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/ImpactedResources_ListBySubscriptionId.json [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        "value": [
+          {
+            "id": "<resourceId>/providers/Microsoft.ResourceHealth/impactedResources/current",
+            "name": "current",
+            "type": "Microsoft.ResourceHealth/impactedResources",
+            "properties": {
+              "availabilityState": "Available",
+              "summary": "Vm is available",
+              "reasonType": "Unplanned",
+              "occuredTime": "2016-03-29T09:12:00Z"
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/ImpactedResources_ListBySubscriptionId.json]:
```suggestion
              "occurredTime": "2016-03-29T09:12:00Z"
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/ImpactedResources_ListBySubscriptionId.json [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+          },
+          {
+            "id": "<resourceId>/providers/Microsoft.ResourceHealth/impactedResources/current",
+            "name": "current",
+            "type": "Microsoft.ResourceHealth/impactedResources",
+            "properties": {
+              "availabilityState": "Unavailable",
+              "summary": "We are sorry, we couldn't automatically recovery our virtualmachine",
+              "reasonType": "Unplanned",
+              "occuredTime": "2016-03-29T09:12:00Z"
```
Comment (nschonni) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/ImpactedResources_ListBySubscriptionId.json]:
```suggestion
              "occurredTime": "2016-03-29T09:12:00Z"
```

**Review by nschonni — COMMENTED** [review]
occured -> occurred

**Review by KrisBash — COMMENTED** [review]
Events looks good. Please see questions regarding impacted resources

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/ResourceHealth.json [hunk 12]
```diff
@@ -0,0 +1,1239 @@
+{
```
[the remark this answers is not in the document]
Comment (srinathgrs) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/ResourceHealth.json]: Sure will do

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/ResourceHealth.json [hunk 13]
```diff
@@ -0,0 +1,1239 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "ListEventsBySingleResource": {
+            "$ref": "./examples/Events_ListBySingleResource.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ResourceHealth/impactedResources": {
```
[the remark this answers is not in the document]
Comment (srinathgrs) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/ResourceHealth.json]: This api is still in preview , I will remove it from the stable version

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/ImpactedResources_ListBySubscriptionId.json [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    "subscriptionId": "subscriptionId",
+    "api-version": "2018-07-01-preview",
+    "$filter": "service eq 'Virtual Machines' or region eq 'West US'"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "<resourceId>/providers/Microsoft.ResourceHealth/impactedResources/current",
```
[the remark this answers is not in the document]
Comment (srinathgrs) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2018-07-01/examples/ImpactedResources_ListBySubscriptionId.json]: Sorry for the confusion, this api is still in preview . I will remove it from this spec

**Comment by srinathgrs** [comment]
I have updated the review based on the review comments , Could you please continue reviewing

**Review by KrisBash — APPROVED** [review]
LGTM thanks

**Comment by srinathgrs** [comment]
> LGTM thanks

Thanks , Looks like i dont have access to merge this pull request . Will it be merged by the ARM team or should i raise a request to get access

**Pull request merged**